### PR TITLE
HPCC-22547 Update uninstall/upgrade package to ensure stoppage of components

### PR DESCRIPTION
**Pull request by Michael-Gardner**
Signed-off-by: Michael Gardner <michael.gardner@lexisnexisrisk.com>
Signed-off-by: Michael Gardner <michael.garder@lexisnexisrisk.com>
Signed-off-by: Michael Gardner <michael.gardner@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by Michael-Gardner** [comment]
@gfortil please review.  Ran your ecl through hthor to spawn an infinite eclagent job to hang in the background.  The killall -9 (SIGKILL) -u hpcc will ensure that any hanging jobs in the background when an admin goes to update the platform will be killed and cleared from memory.  We do try to kill them with a sigterm first using hpcc-init or systemd.  Removed duplicate code from uninstall-init that could cause user confusion because of the output.  Now all shutdown and cleaning of components is contained in the prerm script itself, and uninstall-init is dedicated to removing symlinks laid down in the install-init steps.

**Comment by gfortil** [comment]
tested and works
@richardkchapman ready to be merged

File: initfiles/sbin/prerm.in [hunk 1]
```diff
@@ -28,11 +28,12 @@ fi
 # Stop all services before cleaning up
 ######################################
 if which systemctl 1>/dev/null 2>&1; then
-	systemctl stop hpccsystems-platform.target
+    systemctl stop hpccsystems-platform.target
 else 
 	/etc/init.d/hpcc-init stop
 	/etc/init.d/dafilesrv stop
 fi
+killall -9 -u hpcc
```
Comment (richardkchapman) [inline initfiles/sbin/prerm.in]: Shouldn't the user name be configurable?

But also a bit concerned about the situation where it is - on local installs I run the platform as myself, but I wouldn't want all processes owned by myself killed on removal. Maybe this code is not executed in such cases though

Comment (Michael-Gardner) [reply initfiles/sbin/prerm.in]: The code in the prerm script is not used for 'developer builds' where we change the runtime user and installation location.  That's why file paths and username in this file are hardcoded and it's never been a problem we had to go back and fix. This file only gets embedded in the deb and rpm packages.

Comment (richardkchapman) [reply initfiles/sbin/prerm.in]: The code in install-init.in seems to have user configurable...

Comment (Michael-Gardner) [reply initfiles/sbin/prerm.in]: I'll go ahead and make changes to the uninstall-init, postinst, postrm, and prerm scripts and make sure they work the same way as the install-init script.

**Comment by richardkchapman** [comment]
@Michael-Gardner This is with you I think

**Comment by Michael-Gardner** [comment]
@gfortil Please review the cmake changes I made in the other files.  They're minor and just allow for the configure_file function to do simple replacement, along the lines of what we do for other files.

**Pull request merged**